### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by imjasonh**
Picks up https://github.com/buildpacks/imgutil/pull/124, upgrades the docker/docker dep to v20.10.7, drops pinned x/sys dep

Signed-off-by: Jason Hall <jasonhall@redhat.com>

cc @natalieparellano

### CODE REVIEW

**Comment by imjasonh** [comment]
Also switched to using [module-aware `go install`](https://tip.golang.org/doc/go1.16#modules) to fetch version-pinned tooling dependencies, instead of requiring the import+go.mod hack in tools/tools.go.

I'm not sure if this means tools/go.mod can be dropped entirely, if so I can make that change in this PR as well.

File: tools/go.mod
Comment (jabrown85) [inline tools/go.mod:7]: Can we get rid of `tools/*` completely now that you've upgrade the `Makefile`?

Comment (imjasonh) [reply tools/go.mod]: There are still useful tools in there (`tools/image/main.go`, etc.), but I don't see any reason it has to be a separate Go module anymore. I've pushed another commit that removes `tools/go.*` and reruns `go mod download` to get all of `tools/`'s deps in `/go.mod`.

**Comment by jabrown85** [comment]
Thanks for this! It looks great.

I've approved the CI run. Looks like you need to DCO the latest commit @imjasonh.

**Comment by jabrown85** [comment]
@imjasonh just re-reviewing as CI immediately failed. We need to update to go 1.16 to get this behavior (CI included)

**Comment by jabrown85** [comment]
Linting is failing now - see [here](https://github.com/buildpacks/lifecycle/pull/628/checks?check_run_id=2796025085#step:5:145)

I guess the upgraded linter means new rules and some deprecation of linters entirely

**Comment by imjasonh** [comment]
> Linting is failing now - see [here](https://github.com/buildpacks/lifecycle/pull/628/checks?check_run_id=2796025085#step:5:145)
> 
> I guess the upgraded linter means new rules and some deprecation of linters entirely

Oh my bad, I was blindly upgrading golangci-lint instead of using the same version that had been pinned in tools/go.mod. That should be fixed now 🤞

**Comment by imjasonh** [comment]
Moving tools/* into the main module means it's now subject to lint checks, I've fixed some findings.

**Comment by jabrown85** [comment]
>go install github.com/sclevine/yj@v0.0.0-20200815061347-554173e71934: github.com/sclevine/yj@v0.0.0-20200815061347-554173e71934
	The go.mod file for the module providing named packages contains one or
	more replace directives. It must not contain directives that would cause
	it to be interpreted differently than if it were the main module.

Hmm - I guess we need to add the same replace directive in the main go.mod now or we need @sclevine to cut a version that doesn't have a replace directive (if possible).

**Comment by imjasonh** [comment]
I even tried adding the `replace` directive here, and got the same error message. 😢 

It's possible that module-aware `go install` just doesn't support `replace` in the installed package at all? 🤔

**Comment by jabrown85** [comment]
@imjasonh Stephen pulled in my PR to remove the `replace` directive if you want to pull that in and unblock this PR

**Comment by imjasonh** [comment]
> @imjasonh Stephen pulled in my PR to remove the `replace` directive if you want to pull that in and unblock this PR

Thanks! Done. 👍

**Comment by jabrown85** [comment]
@imjasonh 🤞  the final hurdle appears to be the acceptance testdata launcher Dockerfile(s). The go1.16 tooling expects a go.mod to be present in order to run `go build -o helper <>`. Adding `GO111MODULE=off` to those Dockerfile or adding a go.mod for that exec.d folder should resolve it.

**Comment by imjasonh** [comment]
> @imjasonh 🤞 the final hurdle appears to be the acceptance testdata launcher Dockerfile(s). The go1.16 tooling expects a go.mod to be present in order to run `go build -o helper <>`. Adding `GO111MODULE=off` to those Dockerfile or adding a go.mod for that exec.d folder should resolve it.

Done! It's getting closer I can feel it! 😄

**Comment by jabrown85** [comment]
@imjasonh Linux passed 🎉  Windows 😬 

```
RUN GO111MODULE=off go build -o helper.exe exec.d
         ---> Running in 07384429eba3
        'GO111MODULE' is not recognized as an internal or external command,
```

I think we need something like

`RUN set GO111MODULE=off` before the `RUN go build` command.

File: acceptance/testdata/launcher/Dockerfile.windows
Comment (micahyoung) [inline acceptance/testdata/launcher/Dockerfile.windows:4]: To get this to work, I had to change this:
```suggestion
RUN set "GO111MODULE=off" && go build -o helper.exe exec.d
```

Comment (imjasonh) [reply acceptance/testdata/launcher/Dockerfile.windows]: Thanks!

**Review by jabrown85 — APPROVED** [review]
I'm super happy to see this!

**Comment by natalieparellano** [comment]
I fear that I created these merge conflicts when I merged #617 :'( @imjasonh do you mind resolving them? We could hold off on other merges until this gets through.

**Comment by imjasonh** [comment]
> I fear that I created these merge conflicts when I merged #617 :'( @imjasonh do you mind resolving them? We could hold off on other merges until this gets through.

No worries! Since the conflict was just on `tools/go.sum` which I deleted anyway the merge was very easy. 😈 

I also took this opportunity to squash commits.

**Comment by natalieparellano** [comment]
@imjasonh I think we might have lost the fix for this Windows error in the squash:
```
Step 6/9 : RUN copy c:\gopath\helper.exe c:\layers\0.6_buildpack\some_layer\exec.d\helper.exe
         ---> Running in 01cc5789cf70
        The system cannot find the file specified.
```
Do you remember what it was? If not, I can grab a Windows environment.

**Comment by imjasonh** [comment]
> @imjasonh I think we might have lost the fix for this Windows error in the squash:
> 
> ```
> Step 6/9 : RUN copy c:\gopath\helper.exe c:\layers\0.6_buildpack\some_layer\exec.d\helper.exe
>          ---> Running in 01cc5789cf70
>         The system cannot find the file specified.
> ```
> 
> Do you remember what it was? If not, I can grab a Windows environment.

I'm not sure I understand what changed. The only changes I made here were:

1. Update to Go 1.16, which changed the default `GO111MODULE` to `on`, so I
2. Explicitly `set GO111MODULE=off`

This seems to have caused the `system cannot find the file specified` error, which I've attempted to fix by explicitly writing the built binary to `c:\gopath\helper.exe`. This didn't seem to fix the issue though.

I've pushed another commit that might fix this, but I'm a bit grasping at straws here. If someone knows more about how Windows works that might be helpful.

File: acceptance/testdata/launcher/Dockerfile.windows
Comment (aemengo) [reply acceptance/testdata/launcher/Dockerfile.windows:4]:
```suggestion
WORKDIR /go/src
RUN go build -o helper.exe exec.d
```

Like micahyoung said, something is up with the `;` there. It's not necessary anyway because GO111MODULE=off by default on golang:1.16. And because it's off, you can build first switching to the directory (_as opposed to relying on the c:/gopath_)

File: acceptance/testdata/launcher/Dockerfile.windows
Comment (aemengo) [inline acceptance/testdata/launcher/Dockerfile.windows:5]:
```suggestion
WORKDIR /go/src
RUN go build -o helper.exe exec.d
```

Like micahyoung said, something is up with the `;` there. It's not necessary anyway because GO111MODULE=off by default on golang:1.16. And because it's off, you can build by first switching to the directory (_as opposed to relying on the c:/gopath_)

Comment (imjasonh) [reply acceptance/testdata/launcher/Dockerfile.windows]: Thanks, that makes sense. Let's see if it works 🤞

**Comment by imjasonh** [comment]
The latest failure is:

```
        --- FAIL: TestAnalyzer/acceptance-analyzer/0.3/daemon_case/writes_analyzed.toml (0.52s)
        --- FAIL: TestAnalyzer/acceptance-analyzer/0.3/daemon_case/app_image_exists/restores_app_metadata (62.44s)
```

It only failed on Windows, but looking briefly at the test I'm not sure how that would be platform-dependent, or how it could be related to my change. 🤔 

Any ideas?

**Comment by natalieparellano** [comment]
> seems like a flake?

@imjasonh I'm of that opinion as well. Let's merge this! 

Thank you so much for all the effort in getting this through!

**Pull request merged**
